app/models/edition/timestamps.rb

```diff
@@ -0,0 +1,58 @@
+class Edition::Timestamps
+  def self.edited(
+    edition,
+    payload,
+    previous_live_version = nil,
+    now = Time.zone.now
+  )
+    edition.temporary_first_published_at = previous_live_version&.first_published_at
+    edition.temporary_last_edited_at = now
+    edition.publisher_first_published_at = payload[:first_published_at]
+
+    unless edition.update_type == "major"
+      edition.major_published_at = previous_live_version&.major_published_at
+    end
+
+    edition.first_published_at = payload[:first_published_at] ||
+      previous_live_version&.first_published_at
+
+    edition.last_edited_at = payload[:last_edited_at] || now
+
+    edition.public_updated_at = payload[:public_updated_at] ||
+      previous_live_version&.public_updated_at
+
+    edition.publisher_major_published_at = payload[:public_updated_at]
+    edition.publisher_published_at = nil
+    edition.publisher_last_edited_at = payload[:last_edited_at]
+
+    edition.save!
+  end
+
+  def self.live_transition(
+    edition,
+    previous_live_version = nil,
+    now = Time.zone.now
+  )
+    unless edition.temporary_first_published_at.present?
```

Comment:
I think this would also be a candidate for a comment to explain why always set even if it's not the first item.

Reply:
Just trying to remember the reasoning - was it for data completeness?

Reply:
It's because we always to have a first_published_at value when something has been published. 
We assume lack of one means that something has not been published.
In future we'd like to move this onto the document model so it doesn't have to be passed around so there is less scope for error here.